picocom.c

```diff
@@ -1202,6 +1213,9 @@ loop(void)
 			} while ( n < 0 && errno == EINTR );
 			if ( n <= 0 )
 				fatal("write to term failed: %s", strerror(errno));
+			if ( opts.lecho && opts.log_filename )
+				if ( writen_ni(log_fd, tty_q.buff, sz) < sz )
```

Comment:
No, I'm talking about picocom.c:1217, and the problem is not `tty_q.buff`, but `n` instead of `sz`.